### PR TITLE
Upgrade to JDK 13

### LINKED ISSUE #1906: Upgrade to JDK 13
## JDK 13

"JDK 13, the reference implementation of Java 13, is now Generally Available." as of `2019-09-17`.

https://jdk.java.net/13/

-   350: Dynamic CDS Archives
-   351: ZGC: Uncommit Unused Memory
-   353: Reimplement the Legacy Socket API
-   354: Switch Expressions (Preview)
-   355: Text Blocks (Preview)

## Deliverables

- [ ] Upgrade to JDK 13 as the minimum to build "JUnit 5"
- [ ] ~~Make use of Java [Text Blocks](https://openjdk.java.net/jeps/355) syntax in our test code~~
- [ ] Remove JDK 12 from the build matrix, it reached EOL

### PR DESCRIPTION
**Pull request by sormuras**
## Overview

Resolves #1906

---

I hereby agree to the terms of the [JUnit Contributor License Agreement](https://github.com/junit-team/junit5/blob/002a0052926ddee57cf90580fa49bc37e5a72427/CONTRIBUTING.md#junit-contributor-license-agreement).

---

### Definition of Done

- [ ] There are no TODOs left in the code
- [ ] Method [preconditions](https://junit.org/junit5/docs/snapshot/api/org/junit/platform/commons/util/Preconditions.html) are checked and documented in the method's Javadoc
- [ ] [Coding conventions](https://github.com/junit-team/junit5/blob/master/CONTRIBUTING.md#coding-conventions) (e.g. for logging) have been followed
- [ ] Change is covered by [automated tests](https://github.com/junit-team/junit5/blob/master/CONTRIBUTING.md#tests) including corner cases, errors, and exception handling
- [ ] Public API has [Javadoc](https://github.com/junit-team/junit5/blob/master/CONTRIBUTING.md#javadoc) and [`@API` annotations](https://apiguardian-team.github.io/apiguardian/docs/current/api/org/apiguardian/api/API.html)
- [ ] Change is documented in the [User Guide](https://junit.org/junit5/docs/snapshot/user-guide/) and [Release Notes](https://junit.org/junit5/docs/snapshot/user-guide/#release-notes)
- [ ] All [continuous integration builds](https://github.com/junit-team/junit5#continuous-integration-builds) pass

### CODE REVIEW

**Comment by jbduncan** [comment]
I wonder if the `.lgtm.yml` file should be updated to Java 13 as well. :thinking:

**Comment by marcphilipp** [comment]
Right, and the GitHub actions workflows.

File: settings.gradle.kts [hunk 1]
```diff
@@ -14,8 +14,8 @@ pluginManagement {
 }
 
 val javaVersion = JavaVersion.current()
-require(javaVersion.isJava12Compatible) {
-	"The JUnit 5 build requires Java 12 or higher. Currently executing with Java ${javaVersion.majorVersion}."
+require(javaVersion.isJava13Compatible) {
+	"The JUnit 5 build requires Java 13 or higher. Currently executing with Java ${javaVersion.majorVersion}."
```
Comment (vlsi) [inline settings.gradle.kts]: Oh, man. I just installed Java 12 yesterday to build JUnit, then you upgrade it to 13. That is way too fast :)

Comment (sormuras) [reply settings.gradle.kts]: "12 is dead, Jim."

Just waiting for Gradle 6.0 to be released...

**Comment by sormuras** [comment]
> I wonder if the .lgtm.yml file should be updated to Java 13 as well.

> Right, and the GitHub actions workflows.

Yes and yes.

**Pull request closed without merging**